File: src/main/resources/output/IFC4X3_csharp/Entities/IfcRepresentationItem/IfcGeometricRepresentationItem/IfcLightSource/IfcLightSourcePositional/IfcLightSourceSpot/IfcLightSourceSpot.cs
public class IfcLightSourceSpot : IfcLightSourcePositional
{
    public IfcDirection Orientation { get; set; }
    public IfcReal ConcentrationExponent { get; set; }
    public IfcPositivePlaneAngleMeasure SpreadAngle { get; set; }
    public IfcPositivePlaneAngleMeasure BeamWidthAngle { get; set; }
}

File: src/main/resources/output/IFC4X3_js/Entities/IfcRepresentationItem/IfcGeometricRepresentationItem/IfcLightSource/IfcLightSourcePositional/IfcLightSourceSpot/IfcLightSourceSpot.js
class IfcLightSourceSpot extends IfcLightSourcePositional {
    constructor() {
        /** @type {IFCDIRECTION} */
        this.Orientation = null;
        /** @type {IFCREAL} */
        this.ConcentrationExponent = null;
        /** @type {IFCPOSITIVEPLANEANGLEMEASURE} */
        this.SpreadAngle = null;
        /** @type {IFCPOSITIVEPLANEANGLEMEASURE} */
        this.BeamWidthAngle = null;
    }
}
